### PR TITLE
ClangCompile: add options for the C/C++ standard version

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
This change is [Reviewable](https://reviewable.io/reviews/microsoft/winobjc/2870)

### CODE REVIEW

**Comment by DHowett-MSFT** [comment]
Pre-staging for eventually switching `sdk-build.props` and `ut-build.props` to `Cpp17`!

File: tools/msvc/sbclang.xml [hunk 1]
```diff
@@ -152,7 +152,7 @@
       Category="General"
       DisplayName="Debug Information"
       Description="Specifies whether to keep debug information."
-      Switch="-g"
+      Switch="-g -gcodeview"
```
Comment (DHowett-MSFT) [inline tools/msvc/sbclang.xml:155]: this file is autogenerated!

**Review by MSFTFox — APPROVED** [review]

**Comment by DHowett-MSFT** [comment]
NOTE: This pull request is being built as part of #2871.

**Comment by DHowett-MSFT** [comment]
Merged externally as part of #2869

**Pull request closed without merging**